code/iLasso_for_SINGE.m

```diff
@@ -32,98 +32,111 @@
 numregs = length(pa);
 % Define function for Gaussian kernel
 gkern = @(x,y) gausskernel(x,y,SIG);
+% Check for number of branches, if greater than 1, load branches from the Temporary File
 % Load expression matrix and pseudotime
-X = m.X;
 ptime = m.ptime;
 Xdrop = m.Xdrop;
-rind = (~Xdrop);
+if params.branching==1
+	branches = m.branches;
+else
+	branches = ones(size(ptime));
+end
 
-% Precompute the full kernel matrix -- to be subsampled for each regulator
-if m.computeKp
-    if ismember('fullKp',who(m))&&~isempty(m.fullKp)
-        m.fullKp(1,:) = [];
+for b_ind = 1:min(size(branches))
+    X = m.X;
+    rind = (~Xdrop);
+    
+    % For each branch, only retain the cells corresponding to that branch
+    rind = rind*diag((branches(:,b_ind)>0)*1);
```

Comment:
Do we need error checking in case the values of the branches matrix are outside the expected range?  It seems safe, but I want to double-check the logic in this line.

Reply:
We discuss this and agreed it is okay that the logic checks whether the value is > 0 to select that cell.  We will document it as a 0-1 indicator matrix, however.